packages/module/src/FieldBuilder/FieldBuilder.tsx

```diff
@@ -0,0 +1,330 @@
+ * multi-row forms with a consistent layout. It manages the layout and actions
+ * for adding and removing rows, while giving the consumer full control over the fields themselves.
+ */
+export const FieldBuilder: FunctionComponent<FieldBuilderProps> = ({
+  firstColumnLabel,
+  secondColumnLabel,
+  rowCount,
+  children,
+  onAddRow,
+  onRemoveRow,
+  addButtonProps = {},
+  rowGroupLabelPrefix = "Row",
+  fieldBuilderIdPrefix = "field-builder",
+  ...formGroupProps
+}: FieldBuilderProps) => {
+  // Track the previous row count to detect when rows are added
+  const prevRowCountRef = useRef(rowCount);
+  // Track focusable elements for each row
+  const focusableElementsRef = useRef<Map<number, HTMLElement>>(new Map());
+  // Track remove button refs for focus management
+  const removeButtonRefs = useRef<Map<number, HTMLButtonElement>>(new Map());
+  // Track the add button ref
+  const addButtonRef = useRef<HTMLButtonElement>(null);
+  // ARIA live region for announcing dynamic changes
+  const liveRegionRef = useRef<HTMLDivElement>(null);
+
+  // Generate unique IDs for this instance
+  const instanceId = useRef(`${fieldBuilderIdPrefix}-${Math.random().toString(36).substr(2, 9)}`);
+  const firstColumnLabelId = `${instanceId.current}-first-column-label`;
+  const secondColumnLabelId = secondColumnLabel ? `${instanceId.current}-second-column-label` : undefined;
+
+  // Function to announce changes to screen readers
+  const announceChange = useCallback((message: string) => {
+    if (liveRegionRef.current) {
+      liveRegionRef.current.textContent = message;
+      // Clear the message after a delay to prepare for next announcement
+      setTimeout(() => {
+        if (liveRegionRef.current) {
+          liveRegionRef.current.textContent = '';
+        }
+      }, 1000);
+    }
+  }, []);
+
+  // Create ref callback for focusable elements
+  const createFocusRef = useCallback((rowIndex: number) => 
+    (element: HTMLElement | null) => {
+      if (element) {
+        focusableElementsRef.current.set(rowIndex, element);
+      } else {
+        focusableElementsRef.current.delete(rowIndex);
+      }
+    }, []);
+
+  // Create ref callback for remove buttons
+  const createRemoveButtonRef = useCallback((rowIndex: number) => 
+    (element: HTMLButtonElement | null) => {
+      if (element) {
+        removeButtonRefs.current.set(rowIndex, element);
+      } else {
+        removeButtonRefs.current.delete(rowIndex);
+      }
+    }, []);
+
+  // Enhanced onAddRow with focus management and announcements
+  const handleAddRow = useCallback(() => {
+    onAddRow();
+    announceChange(`New ${rowGroupLabelPrefix.toLowerCase()} added. ${rowGroupLabelPrefix} ${rowCount + 1}.`);
+  }, [ onAddRow, announceChange, rowGroupLabelPrefix, rowCount ]);
+
+  // Enhanced onRemoveRow with focus management and announcements
+  const handleRemoveRow = useCallback((index: number) => {
```

Comment:
For the logic here, I'm thinking that we shouldn't move focus to another destructive action, especially since there's no confirmation when removing a row. Rather than moving to another row's "remove" button, can we move focus to the first input in that row perhaps?

Though another thought I suppose is whether we should be providing this logic or if it should be up to the consumer. We don't handle any logic for adding a row - that's done in the examples. Maybe input from others on whether this sort of logic should be handled by component-groups.